vuu-ui/packages/vuu-layout/src/layout-persistence/LocalLayoutPersistenceManager.ts

```diff
@@ -8,72 +8,124 @@ const metadataSaveLocation = "layouts/metadata";
+          const metadata = (await this.loadMetadata()).filter(metadata => metadata.id !== id);
+          this.saveLayoutsWithMetadata(layouts, metadata);
+          resolve();
+        })
+        .catch(e => reject(e));
+    });
   }
 
-  loadLayout(id: string): LayoutJSON {
-    const layout = this.loadLayouts().filter(layout => layout.id === id);
-
-    switch (layout.length) {
-      case 1: {
-        return layout[0].json;
-      }
-      case 0: {
-        console.log(`WARNING: no layout exists for ID "${id}"; returning empty layout`);
-        return {} as LayoutJSON;
-      }
-      default: {
-        console.log(`WARNING: multiple layouts exist for ID "${id}"; returning first instance`)
-        return layout[0].json;
-      }
-    }
+  loadLayout(id: string): Promise<LayoutJSON> {
+    return new Promise((resolve, reject) => {
+      this.validateId(id, false)
+        .then(async () => {
+          const layouts = (await this.loadLayouts()).filter(layout => layout.id === id);
+          resolve(layouts[0].json);
+        })
+        .catch(e => reject(e));
+    });
   }
 
-  loadMetadata(): LayoutMetadata[] {
-    return getLocalEntity<LayoutMetadata[]>(metadataSaveLocation) || [];
+  loadMetadata(): Promise<LayoutMetadata[]> {
+    return new Promise((resolve) => {
+      const metadata = getLocalEntity<LayoutMetadata[]>(metadataSaveLocation);
+      resolve(metadata || []);
+    })
   }
 
-  private loadLayouts(): Layout[] {
-    return getLocalEntity<Layout[]>(layoutsSaveLocation) || [];
+  private loadLayouts(): Promise<Layout[]> {
+    return new Promise((resolve) => {
+      const layouts = getLocalEntity<Layout[]>(layoutsSaveLocation);
+      resolve(layouts || []);
+    });
   }
 
-  private appendAndPersist(newId: string,
-                           newMetadata: Omit<LayoutMetadata, "id">,
-                           newLayout: LayoutJSON,
-                           existingLayouts: Layout[],
-                           existingMetadata: LayoutMetadata[]) {
-    existingLayouts.push({id: newId, json: newLayout});
-    existingMetadata.push({id: newId, ...newMetadata});
+  private appendAndPersist(
+    newId: string,
+    newMetadata: Omit<LayoutMetadata, "id">,
+    newLayout: LayoutJSON,
+    existingLayouts: Layout[],
+    existingMetadata: LayoutMetadata[]
+  ) {
+    existingLayouts.push({ id: newId, json: newLayout });
+    existingMetadata.push({ id: newId, ...newMetadata });
 
     this.saveLayoutsWithMetadata(existingLayouts, existingMetadata);
   }
 
-  private saveLayoutsWithMetadata(layouts: Layout[], metadata: LayoutMetadata[]): void {
+  private saveLayoutsWithMetadata(
+    layouts: Layout[],
+    metadata: LayoutMetadata[]
+  ): void {
     saveLocalEntity<Layout[]>(layoutsSaveLocation, layouts);
     saveLocalEntity<LayoutMetadata[]>(metadataSaveLocation, metadata);
   }
+
+  private async validateIds(id: string): Promise<void> {
+    return Promise
+      .all([
```

Comment:
could we have just one .catch() on the promise returned by Promise.all() rather than 2 on the individual promises?

Reply:
We can, but it would only report one error (`Promise.all` fails fast, so would jump to the `catch` as soon as one error comes back). I went with this approach to provide a complete error, but we can use a single catch if performance is more of a concern?